src/main/java/htsjdk/tribble/gff/Gff3FeatureImpl.java

```diff
@@ -116,8 +116,8 @@ public Gff3BaseData getBaseData() {
     }
 
     private Set<Gff3FeatureImpl> getDescendents(final Set<String> idsInLineage) {
-        final List<Gff3FeatureImpl> childrenToAdd = children.stream().filter(c -> c.getAttribute(DERIVES_FROM_ATTRIBUTE_KEY) == null ||
-                idsInLineage.contains(c.getAttribute(DERIVES_FROM_ATTRIBUTE_KEY))).
+        final List<Gff3FeatureImpl> childrenToAdd = children.stream().filter(c -> c.getAttribute(DERIVES_FROM_ATTRIBUTE_KEY).isEmpty() ||
+                !Collections.disjoint(idsInLineage, new HashSet<>(c.getAttribute(DERIVES_FROM_ATTRIBUTE_KEY)))).
```

Comment:
I don't think it's necessary to create a set here since it accepts collections and then tries to figure out the fastest way to check them.